### PR TITLE
V1.3 help individualised help

### LINKED ISSUE #158: Integrate `help` into individual commands
Integrate the help feature to individual commands to just provide help for a specified command

### PR DESCRIPTION
**Pull request by appleraincoat**
# Enhance Help
This pull request introduces enhancements to the `help` command, where help for a specified command can be requested.

- Parsing for commands now have an extra check to check if help is being requested for that command, or just an execution of that command.
- New HelpCommandParser class created that parses the variation of help commands (help, add help, delete help etc).

- Help messages may change in future iterations of this PR.

***

## Fixes #158

### CODE REVIEW

File: src/main/java/seedu/realodex/logic/parser/RealodexParser.java [hunk 1]
```diff
@@ -46,36 +46,68 @@ public Command parseCommand(String userInput) throws ParseException {
...
             return new ClearCommand();
 
         case FilterCommand.COMMAND_WORD:
+            if (isHelp) {
+                return new HelpCommandParser().parse(commandWord);
+            }
             return new FilterCommandParser().parse(arguments);
 
         case ListCommand.COMMAND_WORD:
+            if (isHelp) {
```
Comment (UdhayaShan1) [inline src/main/java/seedu/realodex/logic/parser/RealodexParser.java:95]: Nice!

File: src/main/java/seedu/realodex/logic/parser/RealodexParser.java [hunk 2]
```diff
@@ -46,36 +46,68 @@ public Command parseCommand(String userInput) throws ParseException {
...
+                return new HelpCommandParser().parse(commandWord);
+            }
             return new ListCommand();
 
         case ExitCommand.COMMAND_WORD:
+            if (isHelp) {
+                return new HelpCommandParser().parse(commandWord);
```
Comment (UdhayaShan1) [inline src/main/java/seedu/realodex/logic/parser/RealodexParser.java:102]: Perhaps should add tests here

File: src/main/java/seedu/realodex/logic/parser/RealodexParser.java [hunk 3]
```diff
@@ -46,36 +46,68 @@ public Command parseCommand(String userInput) throws ParseException {
...
+            }
             return new ExitCommand();
 
         case HelpCommand.COMMAND_WORD:
-            return new HelpCommand();
+            if (isHelp) {
+                return new HelpCommandParser().parse(commandWord);
+            }
+            return new HelpCommand("");
```
Comment (UdhayaShan1) [inline src/main/java/seedu/realodex/logic/parser/RealodexParser.java:110]:
```suggestion
            return new HelpCommand("");
```

File: src/main/java/seedu/realodex/logic/parser/RealodexParser.java [hunk 4]
```diff
@@ -46,36 +46,68 @@ public Command parseCommand(String userInput) throws ParseException {
...
         case ListCommand.COMMAND_WORD:
+            if (isHelp) {
+                return new HelpCommandParser().parse(commandWord);
+            }
             return new ListCommand();
 
         case ExitCommand.COMMAND_WORD:
+            if (isHelp) {
+                return new HelpCommandParser().parse(commandWord);
+            }
```
Comment (UdhayaShan1) [inline src/main/java/seedu/realodex/logic/parser/RealodexParser.java:103]:
```suggestion
```

Comment (UdhayaShan1) [reply src/main/java/seedu/realodex/logic/parser/RealodexParser.java:103]: No need for them as they are help and exit

File: src/main/java/seedu/realodex/logic/parser/RealodexParser.java (hunk 3 above)
Comment (UdhayaShan1) [reply src/main/java/seedu/realodex/logic/parser/RealodexParser.java:110]: Similarly

**Review by UdhayaShan1 — CHANGES_REQUESTED** [review]
LGTM, thanks do make trivial changes to make CodeCov pass

File: src/main/java/seedu/realodex/logic/parser/RealodexParser.java (hunk 2 above)
Comment (appleraincoat) [reply src/main/java/seedu/realodex/logic/parser/RealodexParser.java:102]: added!

**Pull request closed without merging**